website/notifications/tasks.py

```diff
@@ -0,0 +1,225 @@
+
+        provider = AbstractProvider.objects.get(id=group['provider_id'])
+        additional_context = dict()
+        if isinstance(provider, RegistrationProvider):
+            provider_type = 'registration'
+            submissions_url = get_registration_provider_submissions_url(provider)
+            withdrawals_url = f'{submissions_url}?state=pending_withdraw'
+            notification_settings_url = f'{settings.DOMAIN}registries/{provider._id}/moderation/notifications'
+            if provider.brand:
+                additional_context = {
+                    'logo_url': provider.brand.hero_logo_image,
+                    'top_bar_color': provider.brand.primary_color
+                }
+        elif isinstance(provider, CollectionProvider):
+            provider_type = 'collection'
+            submissions_url = f'{settings.DOMAIN}collections/{provider._id}/moderation/'
+            notification_settings_url = f'{settings.DOMAIN}registries/{provider._id}/moderation/notifications'
+            if provider.brand:
+                additional_context = {
+                    'logo_url': provider.brand.hero_logo_image,
+                    'top_bar_color': provider.brand.primary_color
+                }
+            withdrawals_url = ''
+        else:
+            provider_type = 'preprint'
+            submissions_url = f'{settings.DOMAIN}reviews/preprints/{provider._id}',
+            withdrawals_url = ''
+            notification_settings_url = f'{settings.DOMAIN}reviews/{provider_type}s/{provider._id}/notifications'
+
+        if not rendered_notifications:
+            log_message(f"No notifications to send for user {user._id} with message frequency {message_freq}")
+            continue
+        event_context = {
+            'notifications': ' <br>'.join(rendered_notifications),
+            'user_fullname': user.fullname,
+            'can_change_preferences': False,
+            'notification_settings_url': notification_settings_url,
+            'withdrawals_url': withdrawals_url,
+            'submissions_url': submissions_url,
+            'provider_type': provider_type,
+            'additional_context': additional_context,
+            'is_admin': provider.get_group(ADMIN).user_set.filter(id=user.id).exists()
+        }
+
+        notification_type = NotificationType.objects.get(name=NotificationType.Type.DIGEST_REVIEWS_MODERATORS.value)
+        notification_type.emit(user=user, event_context=event_context, is_digest=True)
+
+        for notification in notifications_qs:
+            notification.mark_sent()
+
+
+def get_moderators_emails(message_freq: str):
+    """Get all emails for reviews moderators that need to be sent, grouped by users AND providers.
+    :param send_type: from NOTIFICATION_TYPES, could be "email_digest" or "email_transactional"
+    :return Iterable of dicts of the form:
+    """
+    sql = """
+        SELECT
```

Comment:
I wonder if this could be re-written using ORM? That makes it a bit more readable, however this is acceptable.